src/primitives/transaction.h

```diff
@@ -14,12 +14,71 @@
 #include "consensus/consensus.h"
 
 #include <boost/array.hpp>
+#include <boost/variant.hpp>
 
 #include "zcash/NoteEncryption.hpp"
 #include "zcash/Zcash.h"
 #include "zcash/JoinSplit.hpp"
 #include "zcash/Proof.hpp"
 
+namespace libzcash {
+    typedef base_blob<1536> GrothProof;
```

Comment:
(48 + 48 + 96)*8 = 1536, ACK!

Reply:
Define a constexpr with the above math and explain what each component is.  Magic numbers should be explained.

Reply:
Agreed.

48 = BLS12-381 G1 element (compressed) when serialized
96 = BLS12-381 G2 element (compressed) when serialized

Groth16 proof = `A : G1` || `B : G2` || `C : G1`

Reply:
Sure, I can add that for now, though it will get removed as soon as we define the actual structures (because the magic numbers will be inherent in the parsed datatypes).